patroni/scripts/wale_restore.py

```diff
@@ -115,8 +136,8 @@ def should_use_s3_to_create_replica(self):
                         cur.execute("SELECT pg_xlog_location_diff(pg_current_xlog_location(), %s)", (backup_start_lsn,))
                         diff_in_bytes = long(cur.fetchone()[0])
             except psycopg2.Error as e:
-                logger.error('could not determine difference with the master location: %s', e)
-                return False
+                logger.exception('could not determine difference with the master location: %s', e)
```

Comment:
logger.exception will print stacktrace, so there is no need to pass `e` parameter.

And we should try to connect to the master even if `no_master` is set.